pyanaconda/modules/subscription/subscription_interface.py

```diff
@@ -91,25 +89,6 @@ def SetSystemPurposeData(self, system_purpose_data: Structure):
         converted_data = SystemPurposeData.from_structure(system_purpose_data)
         self.implementation.set_system_purpose_data(converted_data)
 
-    @property
-    def IsSystemPurposeApplied(self) -> Bool:
-        """Report if system purpose data has been applied.
-
-        We don't differentiate between that installation environment and the target system in this
-        case as we will make sure the system purpose data will always end up on the target system
-        if requested by the user, regardless of where it is initially set.
-        """
-        return self.implementation.is_system_purpose_applied
-
-    def SetSystemPurposeWithTask(self) -> ObjPath:
```

Comment:
I don't think it is a good idea to have a setter with side effects. If you want to go this way, then remove the setter from the DBus API and extend the method `SetSystemPurposeWithTask` with an argument for the system purpose data. If the task succeeds, set the property for the system purpose data. However, this solution might have some new drawbacks.

In general, I am not sure if it is worth it to modify the current solution. You can achieve the same effect in the client by calling the method `SetSystemPurposeWithTask` directly after setting the property `SystemPurposeData`. The DBus API should be straightforward.

Reply:
I don't think having the task set the system purpose property is a good idea - that seems like something that could cause race conditions & complicate setting system purpose data overall.

Yeah, let's revert to the current solution with the expectation of the client having to call `SetSystemPurposeWithTask()` to actually apply the system purpose data that has been set (hopefully this does not cause race condition by itself, we can change it later if it does). 

Looking into the attached subscription parsing, this might actually be needed - activation keys might have system purpose data attached, that override what was set on the system before registration, so we might need to set system purpose data just to the Subscription module property, so that it can be shown in the GUI easily. 

In this case the client (GUI in this case) would check if system is subscribed and would not run the task (changing system purpose of subscribed system does is not a good idea). This subscription stuff just needs to be complicated. :P